captain/parser/bool_parser/expressions/exceptions.py

```diff
@@ -0,0 +1,46 @@
+class InvalidCharacter(Exception):
+    def __init__(self, message=""):
+        self.message = "Invalid character: " + message
+        super().__init__(self.message)
+
+
+class MissingRightParanthesis(Exception):
+    def __init__(self, message=""):
+        self.message = "Missing right paranthesis: " + message
+        super().__init__(self.message)
+
+
+class MissingLeftParanthesis(Exception):
+    def __init__(self, message=""):
+        self.message = "Missing left paranthesis" + message
+        super().__init__(self.message)
+
+
+class InvalidExpression(Exception):
+    def __init__(self, message=""):
+        self.message = "The boolean expressoin is invalid: " + message
+        super().__init__(self.message)
+
+
+class TargetNumberMismatch(Exception):
+    def __init__(self, message=""):
+        self.message = "Mismatch in number of expected targets: " + message
+        super().__init__(self.message)
+
+
+class TestNotRan(Exception):
+    def __init__(self, message=""):
+        self.message = "The test hasn't been ran"
+        super().__init__(self.message)
+
+
+class MatchError(Exception):
+    def __init__(self, message=""):
+        self.message = "Something went wrong during matching: " + message
+        super().__init__(self.message)
+
+
+class InvalidIdentifier(Exception):
+    def __init__(self, message=""):
+        self.message = "Invalid identifier: " + message
+        super().__init__(self.message)
```

Comment:
I feel like a lot of these could just be grouped under a single `ParseError`, you're supplying a message anyway right?

Reply:
I feel like it's better for the dev: when he will see in the console the error message, he can grep to the error type and go to its reference via the lsp. If they were all parse errors, just more annoying to navigate.

Reply:
I think different error type makes sense. Sometimes you may not want to handle the error right away so you propagate it up, then in that case having different error types will help I think